### PR TITLE
PRJ: Auto-select WSL based toolchain when opening a project inside WSL FS

### PR DESCRIPTION
**Pull request by mchernyavsky**
Fixes https://github.com/intellij-rust/intellij-rust/issues/7203

changelog: Auto-select WSL based toolchain when opening a project inside WSL FS

### CODE REVIEW

File: src/211/main/kotlin/org/rust/cargo/toolchain/wsl/RsWslToolchainFlavor.kt [hunk 1]
```diff
@@ -19,42 +20,11 @@ import java.nio.file.Path
...
-                if (!localPath.isDirectory()) continue
-                yield(localPath)
-            }
-
-            for (remotePath in listOf("/usr/local/bin", "/usr/bin")) {
-                val localPath = root.resolve(remotePath)
-                if (!localPath.isDirectory()) continue
-                yield(localPath)
-            }
+            yieldAll(distro.getHomePathCandidates())
```
Comment (vlad20012) [inline src/211/main/kotlin/org/rust/cargo/toolchain/wsl/RsWslToolchainFlavor.kt:27]: Let's remove `sequence {}` and just return `distro.getHomePathCandidates()`

**Review by Undin — CHANGES_REQUESTED** [review]
I've dropped 203 support so the code can be simplified.

Also, it would also be useful to select a proper initial toolchain in project creation dialog (at least when default project location is already known, i.e. in minor IDEs). But I think it should be implemented only after #7217 and this PRs

**Comment by mchernyavsky** [comment]
@Undin Done.

File: src/main/kotlin/org/rust/cargo/toolchain/RsToolchainBase.kt [hunk 2]
```diff
@@ -126,11 +129,24 @@ abstract class RsToolchainBase(val location: Path) {
...
+            val path = projectDir?.toString() ?: return null
+            if (!WslDistributionManager.isWslPath(path)) return null
+
+            val distribution = WslPath.getDistributionByWindowsUncPath(path)
+            val toolchain = distribution
+                ?.getHomePathCandidates()
+                ?.filter { RsToolchainFlavor.getFlavor(it) != null }
+                ?.mapNotNull { RsToolchainProvider.getToolchain(it.toAbsolutePath()) }
+                ?.firstOrNull()
+            if (toolchain != null) return toolchain
```
Comment (Undin) [inline src/main/kotlin/org/rust/cargo/toolchain/RsToolchainBase.kt:143]: Now, if you pass `null` as `projectDir` or pass non-wsl path, we always return null as suggested toolchain

**Review by Undin — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/rust/cargo/toolchain/RsToolchainBase.kt (hunk 2 above)
Comment (mchernyavsky) [reply src/main/kotlin/org/rust/cargo/toolchain/RsToolchainBase.kt]: Fixed.

**Review by Undin — APPROVED** [review]
bors r+

**Pull request merged**